### ISSUE TITLE
Project that has been converted to new 2024.2 format will become corrupted for 2024.2 after being opened in monthly due to being unable to handle changes 2023.11 makes to .yy files in the options folder

### ISSUE THREAD

**Issue by KormexGit** [YoYoGames/GameMaker-Bugs#4539]
### Description
While I was told that you won't be able to open a project that has been converted to the new format in older verions anymore, I don't think it's expected that doing so would cause the project to become permanently corrupted and no longer able to be opened in 2024.2 anymore either.
I'm filing this report from 2023.11 since I can still open my corrupted project in 2023.11, so it's the easiest way I can attach it to the report as I can't open the project in the beta version. This is a bug report for beta despite the version the report was filed from. My beta version:
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/100942509/28bd156e-fc93-4ad0-a575-6460d3b55653)
When the project gets corrupted, a new .yyp and resouce_order file are created that have no name. These are not the issue however - the problem is due to changes made to the .yy files in the options folder. I was able to uncorrupt the project by deleting the options folder in the corrupted project and copy pasting the options folder from a functioning 2024.2 project.
So, I believe the issue here is 2024.2 being unable to re-convert or reset the options files after 2023.11 makes weird changes to them.

What appears when opening in 2023.11:
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/100942509/6617191c-6909-445c-9906-1462a4ee061b)

When attempting to open the project in 2024.2 again:
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/100942509/d8b87c97-6ed1-4c77-8313-cbcb2019de01)

### Steps To Reproduce
Create a new project in the 2024.2 beta, or open an existing project there. Open the project in 2023.11. Now attempt to open it in 2024.2 again. You will get a project load error.

[1e52d0bc-f8f7-41f4-90d8-3f87ed940681](https://api.gamemaker.io/api/github/downloads/1e52d0bc-f8f7-41f4-90d8-3f87ed940681)

**Comment by stuckie** [comment]
We need to add resourceVersion and resourceType fields back in.
resourceVersion should be force to 2.0 for all resources
resourceType should be the same as it was
This will stop older IDEs brute-forcing a parse

We will still have the "$TypeTag":"Version" at the top of the file for the Versioned format to parse as designed

**Comment by stuckie** [comment]
Likely the same as #4574

**Comment by stuckie** [comment]
With fixes in https://github.com/YoYoGames/ProjectTool/pull/41 I've been able to verify that November and back to LTS22 will "load" the project, but give you the warning that it's in read-only mode and direct you to a helpdesk article on what you can do.


![Image](https://github.com/YoYoGames/GameMaker-Bugs/assets/795290/fe0acccc-82ef-40ab-8195-94a3b70f7d7b)

I wonder if we should write a quick article on ProjectTool so they can take their projects in if they want to, with all the caveats of resources being taken out if they didn't exist, and some defaults being reset.

**Comment by Emc1923** [comment]
Verified that reopening a project in  IDE v2024.200.0.483  Runtime v2024.200.0.499, that has been opened with 2023.11 loads properly, however the project will still be corrupted in 2023.11 (as shown in the screenshot above initially). Upon re-opening again I get the read-only warning but same corruption: 

![Image](https://github.com/YoYoGames/GameMaker-Bugs/assets/105791928/06e2e74f-9478-435b-93cc-bd87a7dd440c)

**Comment by stuckie** [comment]
Correct, the project would have been corrupted as the resources would not have had any name tags it could reference. This has now been fixed specifically as part of https://github.com/YoYoGames/YoYoStudio/issues/9476

**Comment by Emc1923** [comment]
Verified as of  IDE v2024.200.0.486  Runtime v2024.200.0.502 projects will open read-only in older IDEs, but not corrupted.
